pkg/route/routeapihelpers/routeapihelpers.go

```diff
@@ -42,3 +44,29 @@ func IngressURI(route *routev1.Route, host string) (*url.URL, *routev1.RouteIngr
 	}
 	return nil, nil, fmt.Errorf("no ingress for host %s in route %s in namespace %s", host, route.ObjectMeta.Name, route.ObjectMeta.Namespace)
 }
+
+// ValidateHost checks that a route's host name satisfies DNS requirements.
+// Unless the allowNonCompliant annotation is set to true, host name must have
+// at least two labels, with each label less than 63 chars from a given character
+// set, and total length less than 253 chars.  Trailing dots are allowed.
+// If allowNonCompliant is set to true, only check that the host satisfies
+// basic DNS subdomain conditions, e.g. length and character set.
+func ValidateHost(host string, allowNonCompliant string) field.ErrorList {
+	result := field.ErrorList{}
+	specPath := field.NewPath("spec")
+
+	// host is not required but if it is set ensure it meets certain DNS requirements
+	if len(host) > 0 {
```

Comment:
I think it would make more sense for the caller to check if `spec.host` is empty because only the caller knows if empty `host` is valid in the caller's particular context.

Reply:
I'll remove it and add a comment.